web/src/js/components/General/EmailField.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import TextField from '@material-ui/core/TextField'
+
+class EmailField extends React.Component {
+  constructor(props) {
+    super(props)
+    this.inputElem = null
+    this.state = {
+      email: null,
+      validationErrorMessage: null,
+    }
+  }
+
+  hasValue() {
+    return !!(
```

Comment:
just curious what the type coercion is for?

Reply:
To get it back to a boolean, but I do think it would technically work without this coercion.